emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/DynamoDBConstants.java

```diff
@@ -86,4 +88,5 @@
   String DEFAULT_AWS_REGION = Regions.US_EAST_1.getName();
 
   int DEFAULT_AVERAGE_ITEM_SIZE_IN_BYTES = 100;
+  String DEFAULT_CAPCITY_FOR_ON_DEMAND = "40000";
```

Comment:
The safe read capacity for on-demand is 10,000 RCU for two reasons. First, all newly created on-demand tables will have at least 4 partitions which yields 12,000 RCU peak (4 partitions * 3000 IOPS/partition * 1 RCU/IOPS = 12,000 RCU). Secondly, while [most regions have a 40,000 RCU table limit](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/AuroraMySQL.Integrating.CloudWatch.html) there are still regions that have only 10,000 RCU/s by default. For the record, the safe write limit for a new on-demand table is ~3,500 WCU with a max limit of 4,000 WCU  (4 partitions * 3000 IOPS/partition * (1 WCU)/(3 IOPS) = 4,000 WCU). 

If we keep it at 40,000 (RCU) then the new on-demand tables will have to go through two rounds of splits during the first export, which can take 10 minutes each or longer and will create 500s along the way. This code must be tested against *a brand new on-demand table* to see how this package will respond - will the tasks fail and cause the whole job to fail due to 500s? If so, we should use a safer, lower limit like 10,000 RCU. Ditto for WCUs but the limit is 4,000 WCU on a new on-demand table. 

If we decide to go with a higher default limit, we can discover the maximum on-demand throughput by [calling DescribeLimits](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/dynamodbv2/AmazonDynamoDBClient.html#describeLimits-com.amazonaws.services.dynamodbv2.model.DescribeLimitsRequest-). I can only imagine the impact of adding a new API call to the emr-dynamodb-connector because all users would have to update their IAM policies to allow the call to be made, but technically this is the correct way to find what the limit for the region is. @taklwu Do we want to call the DescribeLimits API to get the upper limit here? Note that the table limit returned by this API is not related to the default number of partitions for a new on-demand table, which is 4.

Reply:
I tested against a newly created OD table.
The MR connector has a a retry/backoff mechanism which works quite well and handles throttling behind the scenes without failing.

![image](https://user-images.githubusercontent.com/11798914/51601002-f1013400-1f0b-11e9-8533-335638500e43.png)

Reply:
Thank you for testing the change on a new table.

The fact the map reduce job completed successfully without failing is the most important result. The way the consumed capacity rose over time is consistent with how DDB will react in on-demand tables. There will be rounds of splits with 5 to 30 minute delays between each round. I'd say question 1 from above is resolved. Question 2 is a feature improvement for another time. Well done, I have no more reservations about the proposed change.